### PR TITLE
Add referer in error.log

### PR DESCRIPTION
**Pull request by highstrike**
Add referer URL in error log, below Request URL.
If the referal is '/' ignore it because it was a direct visit

### CODE REVIEW

**Comment by dereuromark** [comment]
I like the idea, I always had to hack that in myself for the last 5 years with custom handlers :)
But shouldn't you try to reuse the existing one?

    if ($request) {}

Se the code above your addition.

File: src/Error/BaseErrorHandler.php [hunk 1]
```diff
@@ -297,6 +298,11 @@ protected function _getMessage(Exception $exception)
             if ($request) {
                 $message .= "\nRequest URL: " . $request->here();
             }
+
+            $request = new Request();
```
Comment (ADmad) [inline src/Error/BaseErrorHandler.php]: You could instead get request instance using `Router::getRequest()`

Comment (dereuromark) [inline src/Error/BaseErrorHandler.php]: Thats already done at https://github.com/cakephp/cakephp/pull/8089/files#diff-543eecb5875a400afd68469e05fc933fR297
I would just put the code after the line there, no additional check required.

**Comment by highstrike** [comment]
Updated based on feedback

**Comment by dereuromark** [comment]
No, you didnt move it into the check.
Right now it could fail, as request is not always an object. 

    if ($request) { // inside }

**Comment by ADmad** [comment]
Also instead of calling `$request->env('HTTP_REFERER')` twice save result of first call to a variable and then append to message if not empty.

**Comment by highstrike** [comment]
Thanks for the feedback guys.

Sorry @dereuromark, your note about the check came just as i pushed that commit

Hope everything is fine now.

**Comment by ADmad** [comment]
Please squash the commits into a single commit now :)

**Comment by dereuromark** [comment]
:+1:

**Comment by ionas** [comment]
Thanks!

**Comment by highstrike** [comment]
Tried to squash the 4 commits into 1... not sure if this is the expected result :D

**Comment by dereuromark** [comment]
Not yet. You see 6 commits above, should be 1.

**Comment by highstrike** [comment]
I'm going to leave it as it is before i break it completely :stuck_out_tongue:

**Comment by markstory** [comment]
I can add tests and squash merge this in tonight.

**Comment by highstrike** [comment]
Thanks @markstory!

**Comment by markstory** [comment]
Squashed and merged in cffbda24e4a5cc5ea66d4f48502d097db62acdaa

**Pull request closed without merging**